inference-engine/src/mkldnn_plugin/nodes/ctc_loss.cpp

```diff
@@ -72,257 +74,203 @@ class CTCLossImpl : public ExtLayerBase {
             outputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
 
         const auto& logitsShape = inputs[0]->getTensorDesc().getDims();
-        const auto batchNum = logitsShape[0];
-        const auto maxTime = logitsShape[1];
-        const auto classesNum = logitsShape[2];
+        const size_t batchNum = logitsShape[0];
+        const size_t maxTime = logitsShape[1];
+        const size_t classesNum = logitsShape[2];
 
         int blankIndex = classesNum - 1;
         if (inputs.size() > 4) {
             blankIndex = inputs[4]->cbuffer().as<const int*>()[0];
         }
 
-        std::vector<int> targetD(maxTime);
+        std::vector<int> decodedTargetLenB(batchNum, 0);
+        std::vector<std::vector<int>> targetDB(batchNum);
+        std::vector<std::vector<std::vector<float>>> logProbabilitiesB(batchNum);
+        size_t workAmount2 = 0lu;
+
+        auto threadBody_1 = [&](const int ithr, const int nthr) {
```

Comment:
Does this function really required parallel execution?

Reply:
It is usable in case with multiple batches.